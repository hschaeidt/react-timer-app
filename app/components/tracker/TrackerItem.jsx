import React, { PropTypes } from 'react';
import moment from 'moment';

const TrackerItem = ({ date, ranges }) => {
  const totalDuration = moment.duration();

  const renderRanges = () => ranges.map((range, index) => {
    const to = moment(range.to);
    const from = moment(range.from);

    totalDuration.add(moment.duration(to.diff(from)));

    return (
      <div key={index}>
        from {from.format('LT')} to {to.format('LT')}
      </div>
    );
  });

  return (
    <div>
      {moment(date).format('dddd, MMMM Do YYYY')}
      <div>
        Time ranges:
        {renderRanges()}
      </div>
      <div>
        Total time tracked: {totalDuration.hours()} hours {totalDuration.minutes()} minutes
      </div>
    </div>
  );
};

export const propTypes = {
  date: PropTypes.string.isRequired,
  ranges: PropTypes.arrayOf(
    PropTypes.shape({
      from: PropTypes.string.isRequired,
      to: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

TrackerItem.propTypes = propTypes;

export default TrackerItem;
