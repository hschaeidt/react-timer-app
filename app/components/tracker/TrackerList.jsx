import React, { PropTypes } from 'react';
import TrackerItem, { propTypes as trackerItemProps } from './TrackerItem';

const TrackerList = ({ tracks }) => {
  const renderTrackerItems = () => tracks.map((track, index) => (
    <TrackerItem key={index} {...track} />
  ));

  return (
    <div>
      {renderTrackerItems()}
    </div>
  );
};

TrackerList.propTypes = {
  tracks: PropTypes.arrayOf(
    PropTypes.shape(trackerItemProps),
  ).isRequired,
};

export default TrackerList;
