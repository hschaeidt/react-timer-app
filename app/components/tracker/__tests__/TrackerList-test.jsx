import React from 'react';
import { shallow } from 'enzyme';
import TrackerList from '../TrackerList';
import TrackerItem from '../TrackerItem';
import tracks from '../../../fixtures/tracks.json';

function setup() {
  const props = {
    tracks,
    onNewRange: jest.fn(),
    onComplete: jest.fn(),
  };

  const wrapper = shallow(<TrackerList {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('<TrackerList />', () => {
  it('should render one TrackerItem component for each track item', () => {
    const { wrapper } = setup();
    expect(wrapper.find(TrackerItem).length).toBe(tracks.length);
  });
});
