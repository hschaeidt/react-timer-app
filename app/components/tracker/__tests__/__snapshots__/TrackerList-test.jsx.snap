exports[`<TrackerList /> renders correctly 1`] = `
<div>
  <div>
    Friday, December 2nd 2016
    <div>
      Time ranges:
      <div>
        from 
        8:57 AM
         to 
        1:01 PM
      </div>
      <div>
        from 
        2:00 PM
         to 
        6:00 PM
      </div>
    </div>
    <div>
      Total time tracked: 
      8
       hours 
      4
       minutes
    </div>
  </div>
  <div>
    Saturday, December 3rd 2016
    <div>
      Time ranges:
      <div>
        from 
        10:00 AM
         to 
        1:00 PM
      </div>
      <div>
        from 
        2:00 PM
         to 
        7:00 PM
      </div>
    </div>
    <div>
      Total time tracked: 
      8
       hours 
      0
       minutes
    </div>
  </div>
</div>
`;
